sklearn/utils/__init__.py

```diff
@@ -1184,19 +1185,27 @@ def is_abstract(c):
 
     all_classes = []
     # get parent folder
-    path = sklearn.__path__
-    for importer, modname, ispkg in pkgutil.walk_packages(
-            path=path, prefix='sklearn.', onerror=lambda x: None):
-        if ".tests." in modname or "externals" in modname:
-            continue
-        if IS_PYPY and ('_svmlight_format' in modname or
-                        'feature_extraction._hashing' in modname):
-            continue
-        # Ignore deprecation warnings triggered at import time.
-        with ignore_warnings(category=FutureWarning):
-            module = __import__(modname, fromlist="dummy")
-        classes = inspect.getmembers(module, inspect.isclass)
-        all_classes.extend(classes)
+    modules_to_ignore = set(["tests", "externals", "setup"])
+    root = str(Path(__file__).parent.parent)
+    # Ignore deprecation warnings triggered at import time and from walking
+    # packages
+    with ignore_warnings(category=FutureWarning):
+        for importer, modname, ispkg in pkgutil.walk_packages(
+                path=[root], prefix='sklearn.'):
+            mod_parts = modname.split(".")
+            if modules_to_ignore & set(mod_parts) or "._" in modname:
```

Comment:
I'd find using `any` slightly more readable here, instead of set intersection

Reply:
Like this?

```py
if any(part in modules_to_ignore for part in mod_parts)
```

Reply:
yes